File: clases 25-11-2025/Tipos de clases js/clase normal con interfaz/Clases/Estudiante.js

import IEstudiante from '../interfaces/IEstudiante.js';

export default class Estudiante extends IEstudiante {
    constructor(nombre, grado) {
        super();                 
        this.nombre = nombre;
        this.grado = grado;
    }

    mostrarInfo() {
        console.log(`Nombre: ${this.nombre}, Grado: ${this.grado}`);
    }
}

File: clases 25-11-2025/Tipos de clases GO/clase_normal_con_interfaz/Estudiante.go
package main

import "fmt"

type IEstudiante interface {
	MostrarInfo()
}

type Estudiante struct {
	nombre string
	grado  int
}

func NewEstudiante(nombre string, grado int) IEstudiante {
	return Estudiante{nombre: nombre, grado: grado}
}

func (e Estudiante) MostrarInfo() {
	fmt.Printf("Nombre: %s, Grado: %d\n", e.nombre, e.grado)
}
